pyiceberg/table/__init__.py

```diff
@@ -529,6 +529,13 @@ def location(self) -> str:
         """Return the table's base location."""
         return self.metadata.location
 
+    @property
+    def last_sequence_number(self) -> int:
+        return self.metadata.last_sequence_number
+
+    def next_sequence_number(self) -> int:
```

Comment:
Minor: In Java, this is on TableMetadata, which is an internal class and not part of the public API. I don't think anyone would call this externally, but if they did it may not be accurate because the sequence number could be reassigned for a new snapshot if the commit needs to be retried. I'd probably make this `_next_sequence_number()` to avoid that.